src/lambda/models/samLambdaRuntime.ts

```diff
@@ -106,6 +128,25 @@ export function getFamily(runtime: string): RuntimeFamily {
     return RuntimeFamily.Unknown
 }
 
+function handleDeprecatedRuntime(runtime: Runtime) {
+    const moreInfo = localize('AWS.generic.message.learnMore', 'Learn More')
+    vscode.window
+        .showErrorMessage(
+            localize(
+                'AWS.samcli.deprecatedRuntime',
+                'Runtime {0} has been deprecated. Please update to a currently-supported runtime.',
+                runtime
+            ),
+            moreInfo
+        )
+        .then(button => {
+            if (button === moreInfo) {
+                vscode.env.openExternal(vscode.Uri.parse(supportedLambdaRuntimesUrl))
+            }
+        })
+    throw new Error(`Runtime ${runtime} is deprecated, see ${supportedLambdaRuntimesUrl} for more details.`)
```

Comment:
https://github.com/aws/aws-toolkit-vscode/blob/master/docs/CODE_GUIDELINES.md

```suggestion
    throw new Error(`Runtime ${runtime} is deprecated, see: ${supportedLambdaRuntimesUrl}`)
```